js/src/edit-block/components/editor.js

```diff
@@ -86,47 +95,60 @@ const Editor = ( { onError, postId, postType, settings } ) => {
 	const isSavingPost = useSelect( ( select ) => select( 'core/editor' ).isSavingPost() );
 
 	useEffect( () => {
-		if ( isSavingPost && ! block.name ) {
-			const defaultBlock = getDefaultBlock( postId );
-			changeBlockName( defaultBlock.name, defaultBlock );
+		if ( isSavingPost ) {
+			setDefaults( postId );
 		}
-	}, [ block, changeBlockName, isSavingPost, postId ] );
+	}, [ isSavingPost, postId, setDefaults ] );
 
 	if ( ! post ) {
 		return null;
 	}
 
 	return (
 		<StrictMode>
-			<UnsavedChangesWarning />
 			<div className="h-screen flex flex-col items-center text-black">
 				<BrowserURL />
+				<UnsavedChangesWarning />
 				<EditorProvider
 					post={ post }
 					settings={ settings }
 				>
 					<ErrorBoundary onError={ onError }>
-						<EditorNotices />
 						<Header />
+						<EditorNotices />
 						<div className="flex w-full h-0 flex-grow">
-							<Main
-								currentLocation={ currentLocation }
-								selectedField={ selectedField }
-								setCurrentLocation={ setCurrentLocation }
-								setIsNewField={ setIsNewField }
-								setPanelDisplaying={ setPanelDisplaying }
-								setSelectedField={ setSelectedField }
-							/>
+							<Main>
```

Comment:
This is mainly to avoid passing props through so many components.

Instead, it makes component children of other components.